rasa_sdk/executor.py

```diff
@@ -176,54 +182,135 @@ def register_function(self, name: Text, f: Callable) -> None:
-        :type package: str | module
-        :rtype: dict[str, types.ModuleType]
+        Args:
+            package: Package or module name, or an already loaded Python module.
+            recursive: If `True`, and `package` is a package, import all of its
+                sub-packages as well.
         """
         if isinstance(package, str):
-            package = importlib.import_module(package)
+            package = self._import_module(package)
 
         if not getattr(package, "__path__", None):
             return
 
-        results = {}
         for loader, name, is_pkg in pkgutil.walk_packages(package.__path__):
             full_name = package.__name__ + "." + name
-            results[full_name] = importlib.import_module(full_name)
+            self._import_module(full_name)
+
             if recursive and is_pkg:
                 self._import_submodules(full_name)
 
+    def _import_module(self, name: Text) -> types.ModuleType:
+        """Import a Python module. If possible, register the file where it came from.
+
+        Args:
+            name: The module's name using Python's module path syntax.
+
+        Returns:
+            The loaded module.
+        """
+        module = importlib.import_module(name)
+
+        module_file = getattr(module, "__file__", None)
+        if module_file:
+            # If the module we're importing is a namespace package (a package
+            # without __init__.py), then there's nothing to watch for the
+            # package itself.
+            timestamp = os.path.getmtime(module_file)
+            self._modules[module_file] = (timestamp, module)
+
+        return module
+
     def register_package(self, package: Union[Text, types.ModuleType]) -> None:
-        from rasa_sdk.interfaces import Action
+        """Register all the `Action` subclasses contained in a Python module or
+        package.
 
+        Args:
+            package: Module or package name, or an already loaded Python module.
+        """
         try:
             self._import_submodules(package)
         except ImportError:
             logger.exception(f"Failed to register package '{package}'.")
             sys.exit(1)
 
+        self._register_all_actions()
+
+    def _register_all_actions(self) -> None:
+        """Scan for all user subclasses of `Action`, and register them.
+        """
+        import inspect
+
         actions = utils.all_subclasses(Action)
 
         for action in actions:
-            meta = action.__dict__.get("Meta", False)
-            abstract = getattr(meta, "abstract", False)
             if (
                 not action.__module__.startswith("rasa_core.")
                 and not action.__module__.startswith("rasa.")
                 and not action.__module__.startswith("rasa_sdk.")
                 and not action.__module__.startswith("rasa_core_sdk.")
-                and not abstract
+                and not inspect.isabstract(action)
```

Comment:
how does this function work? :-D

Reply:
don't we need to do `Action(ABC)` then or use `@abstractmethod`??

Reply:
Yes! I changed it because that's the canonical way of making abstract classes in Python. I wasn't able to find why the previous method was being used.

Reply:
@federicotdn Did you do this one yet?

Reply:
What would you change here? The project itself does not have any abstract classes that I could find 🤔

Reply:
ah, true. `class Action` is filtered out by the other checks anyway